certstore/certstore.go

```diff
@@ -0,0 +1,120 @@
+package certstore
+
+import (
+	"time"
+
+	"crypto/x509"
+	"database/sql"
+
+	cferr "github.com/cloudflare/cfssl/errors"
+	"github.com/cloudflare/cfssl/log"
+	_ "github.com/mattn/go-sqlite3" // Initialize SQLite without bringing it in scope
+)
+
+// Certificate represents a certificate in the certstore
+type Certificate struct {
+	AsPEM            string
+	Serial           string
+	SignedAt         time.Time
+	Expiration       time.Time
+	RevokedAt        *time.Time
+	RevocationReason sql.NullInt64
+}
+
+var certificatesSchema = `
+	CREATE TABLE IF NOT EXISTS certificates (
+		asPEM VARCHAR(10000),
+		serial VARCHAR(50) PRIMARY KEY,
+		signedAt TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+		expiration TIMESTAMP,
+		revokedAt TIMESTAMP NULL DEFAULT NULL,
+		revocationReason INTEGER NULL DEFAULT NULL
+	);
+`
+
+func initializeDB() (db *sql.DB, err error) {
+	db, err = sql.Open("sqlite3", "./certstore.db")
+
+	if err != nil {
+		return nil, cferr.New(cferr.CertStoreError, cferr.DatabaseInitializationFailed)
+	}
+
+	_, err = db.Exec(certificatesSchema)
+	return db, err
+}
+
+// RecordCert records a new signed cert in the certificate store
+func RecordCert(cert x509.Certificate, signedCert []byte) (err error) {
+	db, err := initializeDB()
```

Comment:
I think we can skip this, and get db *sql.DB as part of the pass-in arguments. We just assume db works. Keep the logic of connecting to db separated.